src/OrleansCodeGenerator/GrainReferenceGenerator.cs

```diff
@@ -191,7 +192,8 @@ private static MemberDeclarationSyntax[] GenerateInvokeMethods(Type grainType, A
                 var options = GetInvokeOptions(method);
 
                 // Construct the invocation call.
-                if (method.ReturnType == typeof(void))
+                var isOneWayTask = method.GetCustomAttribute<OneWayAttribute>() != null;
```

Comment:
You're right - that would be better. If we decide that this PR is desirable then I'll make that change.